modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/AbstractKeyValueStorage.java

```diff
@@ -196,6 +223,41 @@ public long getCompactionRevision() {
         }
     }
 
+    @Override
+    public void startCompaction(long revision) {
+        assert revision >= 0 : revision;
+
+        rwLock.writeLock().lock();
+
+        try {
+            assertCompactionRevisionLessThanCurrent(revision, rev);
+
+            if (isRecoveryState()) {
+                compactionRevision = revision;
+            } else {
+                watchProcessor
+                        .addTaskToWatchEventQueue(() -> setCompactionRevision(revision))
+                        .thenComposeAsync(unused -> readOperationsFuture(revision), compactionExecutor)
+                        .thenRunAsync(() -> compact(revision), compactionExecutor)
+                        .whenComplete((unused, throwable) -> {
+                            if (throwable == null) {
+                                log.info("Metastore compaction completed successfully: [compactionRevision={}]", revision);
+                            } else {
+                                log.error(
+                                        "Metastore compaction completed unsuccessfully: [compactionRevision={}]",
```

Comment:
```suggestion
                                        "Metastore compaction failed: [compactionRevision={}]",
```

Reply:
`completed unsuccessfully` reads really strange and can be confused with the successful outcome

Reply:
Why not.